src/Uno.UI.RemoteControl.Host/RemoteControlServer.cs

```diff
@@ -2,64 +2,110 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Net.WebSockets;
+using System.Reflection;
+using System.Runtime.Loader;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
 using Newtonsoft.Json;
+using Uno.Extensions;
 using Uno.UI.RemoteControl.Helpers;
 using Uno.UI.RemoteControl.HotReload.Messages;
-using Microsoft.Extensions.Logging;
-using Uno.Extensions;
 using Uno.UI.RemoteControl.Messages;
-using System.Runtime.Loader;
-using Microsoft.Extensions.Configuration;
 
 namespace Uno.UI.RemoteControl.Host
 {
 	internal class RemoteControlServer : IRemoteControlServer, IDisposable
 	{
-		private readonly Dictionary<string, IServerProcessor> _processors = new Dictionary<string, IServerProcessor>();
+		private readonly static Dictionary<string, (AssemblyLoadContext Context, int Count)> _loadContexts = new();
+		private readonly Dictionary<string, IServerProcessor> _processors = new();
 
-		private System.Reflection.Assembly? _loopAssembly;
+		private string? _resolveAssemblyLocation;
 		private WebSocket? _socket;
+		private readonly List<string> _appInstanceIds = new();
```

Comment:
Can't we use `_loadContexts` to get the list of appInstanceIds? This would also avoid to have to synchronize both collections (it's enumerated from multiple threads in Dispose)

Reply:
the _appInstanceIds are specific to the RCServer instance. These are tracked separately so we know what to decrement and possibly remove from the (shared) _loadContexts.